core/sql/optimizer/RelRoutine.cpp

```diff
@@ -1081,33 +1085,37 @@ NARoutine * PredefinedTableMappingFunction::getRoutineMetadata(
      BindWA *bindWA)
 {
   NARoutine *result = NULL;
+  ComRoutineLanguage lang = COM_LANGUAGE_CPP;
+  ComRoutineParamStyle paramStyle = COM_STYLE_CPP_OBJ;
+  const char *externalName = NULL;
+  // By default, predefined UDRs share a DLL.
+  const char *libraryFileName = "libudr_predef.so";
   NAString libraryPath;
 
   // the libraries for predefined UDRs are in the regular
   // library directory $MY_SQROOT/export/lib${SQ_MBTYPE}
   libraryPath += getenv("MY_SQROOT");
   libraryPath += "/export/lib";
-  libraryPath += getenv("SQ_MBTYPE");
 
   switch (getOperatorType())
     {
     case REL_TABLE_MAPPING_BUILTIN_LOG_READER:
+      externalName = "TRAF_CPP_EVENT_LOG_READER";
+      libraryPath += getenv("SQ_MBTYPE");
+      break;
+
     case REL_TABLE_MAPPING_BUILTIN_TIMESERIES:
-      {
-        // produce a very simple NARoutine, most of the
-        // error checking and determination of output
-        // columns is done by the compiler interface of
-        // this predefined table mapping function
-        result = new(bindWA->wHeap()) NARoutine(routineName,
-                                                bindWA->wHeap());
-        if (getOperatorType() == REL_TABLE_MAPPING_BUILTIN_LOG_READER)
-          result->setExternalName("TRAF_CPP_EVENT_LOG_READER");
-        else
-          result->setExternalName("TRAF_CPP_TIMESERIES");
-        result->setLanguage(COM_LANGUAGE_CPP);
-        result->setRoutineType(COM_TABLE_UDF_TYPE);
-        result->setParamStyle(COM_STYLE_CPP_OBJ);
-      }
+      externalName = "TRAF_CPP_TIMESERIES";
+      libraryPath += getenv("SQ_MBTYPE");
+      break;
+
+    case REL_TABLE_MAPPING_BUILTIN_JDBC:
+      lang = COM_LANGUAGE_JAVA;
+      paramStyle = COM_STYLE_JAVA_OBJ;
+      externalName = "org.trafodion.sql.udr.predef.JDBCUDR";
+      libraryPath += "/trafodion-sql-";
```

Comment:
Why do we add a slash for libraryPath in this case but not in the previous two cases? (Is there something different about slashes for C++ and Java?)